tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts

```diff
@@ -0,0 +1,217 @@
+      x: rightIndex * distanceBetweenPoints,
+      offset: 0,
+      isClicked: false,
+      index: rightIndex,
+      isExist: rightIndex !== -1,
+      isSelected: true,
+    };
+    this.moveBridge();
+  }
+
+  // When user clicks on drag element
+  mousedown(event: MouseEvent, dragElement: DragElement): void {
+    if (event.which === 1) { // left mouse button
+      if (this.menuOffset === undefined) {
+        // We gets mouse x on whole screen, but we need x on menu component only
+        this.menuOffset = event.pageX - event.layerX - dragElement.x;
+      }
+      dragElement.offset = event.offsetX;
+      dragElement.isClicked = true;
+    }
+  }
+
+  // Moves drag element to nearest point and gets commit id of the point
+  getCommitId(
+    dragElement: DragElement,
+    dragElement2: DragElement,
+    step: number,
+  ): string {
+    let index: number = this.getNearestX(dragElement.x);
+    if (index === dragElement2.index && dragElement2.isExist) {
+      // If nearest point is already occupied
+      index += step;
+    }
+    dragElement.index = index;
+    dragElement.x = distanceBetweenPoints * index;
+    this.moveBridge();
+    return this.stateService.commitIdList[index];
+  }
+
+  // Converts mouse X to X of drag element
+  getX(mouseX: number, dragElement: DragElement): number {
+    let x: number = mouseX - this.menuOffset - dragElement.offset;
+    x = Math.max(x, 0);
+    x = Math.min(x, (this.stateService.commitIdList.length - 1) * distanceBetweenPoints);
+    return x;
+  }
+
+  // Pulls drag element to nearest point
+  magnet(x: number, dragElement: DragElement): void {
+    const centerX: number = x + dragWidth / 2;
+    const n: number = this.getNearestX(x);
+    const pointX: number = dragWidth / 2 + distanceBetweenPoints * n;
+    if (centerX > pointX - magnetField / 2 && centerX < pointX + magnetField / 2) {
+      dragElement.x = pointX - dragWidth / 2;
+      dragElement.isSelected = true;
+    } else {
+      dragElement.isSelected = false;
+      dragElement.x = x;
+    }
+  }
+
+  // Gets X of nearest point to the x
+  getNearestX(x: number): number {
+    return Math.round(x / distanceBetweenPoints);
+  }
+
+  moveDragElement(
+    mouseX: number,
+    dragElement: DragElement,
+    dragElement2: DragElement,
+    step: number,
+  ): void {
+    let x: number = this.getX(mouseX, dragElement);
+
+    // Stop drag element by second drag element
+    if (dragElement2.isExist) {
+      const secondDragFrameX: number = dragElement2.x + dragWidth * step;
+      if (x * step < secondDragFrameX * step) {
+        x = secondDragFrameX;
+      }
+    }
+
+    this.magnet(x, dragElement);
+  }
+
+  moveBridge(): void {
+    this.bridge.x = this.leftDrag.x + dragWidth / 2;
+    this.bridge.width = this.rightDrag.x - this.leftDrag.x;
+  }
+
+  showPopup(commitIndex: number): void {
+    this.commitInfo.id = this.stateService.commitIdList[commitIndex];
+    // TODO: Display real time of a commit
```

Comment:
Can you open an issue for this one, with a P2?

Reply:
:+1: https://github.com/google/startup-os/issues/425